common/internal-types/src/lib.rs

```diff
@@ -15,6 +15,9 @@ pub mod errors;
 /// Types related to internal HOPR protocol logic.
 pub mod protocol;
 
+#[doc(hidden)]
+pub mod legacy; // TODO: remove this in 3.0
```

Comment:
Please, create a "Deprecate features in 3.0.0" issue and link the issue created from this TODO along with https://github.com/hoprnet/hoprnet/issues/5901 there.

Reply:
See https://github.com/hoprnet/hoprnet/issues/6158